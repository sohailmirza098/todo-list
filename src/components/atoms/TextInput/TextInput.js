// import React from 'react'

// export default function TextInput(props) {
//     return (
//         <input data-testid="texttodo" type="text" name="title" placeholder="Add Todo..." style={{flex:'10', padding: '5px'}} value={props.title} onChange={props.onChange} autoComplete='off' />             
//     )
// }


import React from 'react'

const TextInput = (props) => {
    return(
        <input data-testid="texttodo" type="text" name="title" placeholder="Add Todo..." style={{flex:'10', padding: '5px'}} value={props.title} onChange={props.onChange} autoComplete='off' />
    );
}

export default TextInput; 