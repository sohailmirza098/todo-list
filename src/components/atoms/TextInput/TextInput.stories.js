import React from 'react';
import TextInput from "./TextInput";
import { action } from '@storybook/addon-actions';
// import { text } from '@storybook/addon-knobs/react';

export default {
  title: 'TextInput',
  component: TextInput,
};

export const input = () => {
    return(<TextInput />);
};
