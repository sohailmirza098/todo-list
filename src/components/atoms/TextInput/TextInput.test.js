import React from "react";
import { create } from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import TextInput from "./TextInput";

describe("Remove Button testing", () => {

  test("Matches the snapshot", () => {
    const input = create(<TextInput />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("Textbox is present", () => {
    const {getByTestId, getByText} =  render(<TextInput />);
    const text = getByTestId("texttodo");
    expect(text).toBeInTheDocument();
  });

  test("Textbox placeholder", () => {
    const {getByTestId, getByPlaceholderText} =  render(<TextInput />);
    const text = getByPlaceholderText("Add Todo...");
    expect(text.placeholder).toEqual("Add Todo...")
  });

//   test("RemoveButton has content X", () => {
//     const {getByRole}=  render(<RemoveButton />);
//     const button = getByRole("button");
//     expect(button).toHaveTextContent("x");
//   });

//   test("RemoveButton onclick",()=>{
//     const mockClickMethod = jest.fn();
//     const {getByRole}=  render(<RemoveButton onClick={mockClickMethod()}/>);
//     fireEvent.click(getByRole("button"));
//     expect(mockClickMethod).toHaveBeenCalledTimes(1);
//   });
});