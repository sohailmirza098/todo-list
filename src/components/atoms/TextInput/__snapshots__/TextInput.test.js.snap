exports[`Remove Button testing Matches the snapshot 1`] = `
<input
  autoComplete="off"
  data-testid="texttodo"
  name="title"
  onChange={undefined}
  placeholder="Add Todo..."
  style={
    Object {
      "flex": "10",
      "padding": "5px",
    }
  }
  type="text"
  value={undefined} />
`;
